src/ln/peer_handler.rs

```diff
@@ -882,7 +909,7 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 									continue
 								}
 								peer.pending_outbound_buffer.push_back(peer.channel_encryptor.encrypt_message(&encoded_msg[..]));
-								Self::do_attempt_write_data(&mut (*descriptor).clone(), peer);
+								self.do_attempt_write_data(&mut (*descriptor).clone(), peer);
```

Comment:
Note that you have to now check that the update isn't below the last_synced_channel here (and for channel_announcement/node_announcement broadcasts).

Reply:
I dont quite follow which update?

Reply:
When we broadcast new updates we just received, we cant broadcast them to peers which we haven't yet sent the channel_announcement to. Further, even for channel_announcements we dont want to send them via the normal process to a syncing peer until the syncing peer has caught up to their position in the sync process to avoid needlessly double-sending the same announcements.

Reply:
If we disable this while sending, the peer might miss some updates 
Peer might have received announcement from us already. And by disabling this he might miss it from us, but he might get this from a non syncing node. 
If we dont disable, we might double send updates, and he might through away updates.

I think disable is the better one of the two. But we might cache those in a separate list and send them out after he sync's?

Reply:
It shouldn't miss any updates - because we're loading the latest data from the Router as we iterate through the short_id list we should simply send them the latest copy when we get there and never send them the old data.

Reply:
But what if that we already sent them that update? We need to check for that, if we already sent that channel update we need to sent the latest one